src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,73 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+
+        return storageDao.add(user);
+    }
+
+    private void validateLoginUniqueness(String login) {
+        if (storageDao.get(login) != null) {
+            throw new UserDataInvalidException("User with such login already exists");
+        }
+    }
+
+    private void validateLogin(String login) {
+        if (login.length() < MIN_LOGIN_LENGTH) {
+            throw new UserDataInvalidException("User`s login length should contain at least"
+                    + MIN_LOGIN_LENGTH
+                    + "characters");
```

Comment:
The exception message should include the actual login length to provide more informative feedback. For example: "User's login length should contain at least " + MIN_LOGIN_LENGTH + " characters, got: " + login.length() + "."

Reply:
changed